doc/tutorial/tutorial.md

```diff
@@ -1497,6 +1497,26 @@ BestDeal(best) :-
     var best = (vendor, price).group_by(item).best_vendor_string().
 ```
 
+Similar to vectors, sets, and other container types, groups are iterable
+objects, and can be iterated over in a for-loop of flattened by a `FlatMap`
+operator.  Let us have a closer look at the for-loop in the last code snippet:
+
+```
+for (((vendor, price), w) in g) { .. }
+```
+
+The iterator yields a 2-tuple, where the first component `(vendor, price)` of
```

Comment:
This is tied to projection. You can implement projection in multiple ways: return a tuple multiple times with different weights, or return it once with the sum of the weights. 

For relations the internal weights should really be invisible to users, they are an artifact of the differential implementation (logically the set should be distincted before being passed to group-by, and then the weights would all 1). For multisets the weights will be important.

Reply:
In fact, I believe that for a relation you should apply "distinct" to the weight before you pass it to the group function, i.e., `w -> if (w>0) then 1 else 0`. For a stream or multiset you should pass the real weight.